### ISSUE TITLE
The PestAdapter builds the PHPUnit configuration with the pest version instead of the PHPUnit version

### ISSUE THREAD

**Issue by MOuli90** [infection/infection#1795]
| Question    | Answer
| ------------| ---------------
| Infection version | 0.26.16
| Test Framework version | PHPUnit 9.5.27 and pestphp/pest 1.22.3
| PHP version | 8.1.14
| Platform    | Doesn't matter
| Github Repo | -

I am running infection with the test runner `pestphp/pest`. While I was looking for a strange behaivor I could find a small (not intended?) bug. The PestAdapter uses the pest version instead of the PHPUnit version to create the phpunit config files. 

The partial stacktrace should be a prove
```
#0 ...\vendor\infection\infection\src\TestFramework\AbstractTestFrameworkAdapter.php(111): Infection\TestFramework\AbstractTestFrameworkAdapter->retrieveVersion()
#1 ...\vendor\infection\infection\src\TestFramework\AbstractTestFrameworkAdapter.php(121): Infection\TestFramework\AbstractTestFrameworkAdapter->getVersion()
#2 ...\vendor\infection\infection\src\TestFramework\AbstractTestFrameworkAdapter.php(76): Infection\TestFramework\AbstractTestFrameworkAdapter->buildInitialConfigFile()
#3 ...\vendor\infection\infection\src\TestFramework\PhpUnit\Adapter\PhpUnitAdapter.php(108): Infection\TestFramework\AbstractTestFrameworkAdapter->getInitialTestRunCommandLine('--coverage-xml=...', Array, false)
#4 ...\vendor\infection\infection\src\TestFramework\PhpUnit\Adapter\PestAdapter.php(89): Infection\TestFramework\PhpUnit\Adapter\PhpUnitAdapter->getInitialTestRunCommandLine('--coverage-xml=...', Array, false)
...
```
This will result in calling `pest --version` with the output:
```
Pest    1.22.3
PHPUnit 9.5.27 by Sebastian Bergmann and contributors.
```
an results into `1.22.3` wich is totaly correct, but not usable for the MutationConfigBuilder.

TLDR: The configuration for a unit test will not include any PHPUnit Version related configuration adjustments like enabling result cache, set the execution order etc.

Let me know if i have to share more informations

**Comment by MOuli90** [comment]
I see two posible solutions, maybe you have some more as maintainer ;)

1. Creating a configuration builder for Pest 
 - should be possible but maybe not the best solution -> pest could require diffrent PHPUnit Versions, so the "Version match logic" could be a little bit strage
 2. Use two diffrent Version "reader" one for the output of the current pest version and one for the current PHPUnit version, that should be used to build the configuration. I think this should be the prefered one

**Comment by maks-rafalko** [comment]
I see the bug. However, I don't think changing the very base interface (`TestFrameworkAdapter`) or abstract class is a good way, because Pest is the only one exception that works in this way, relying internally on other test framework adapter.

The third alternative solution I had in my mind after some thinking is to update `PestFrameworkAdapter` here 

https://github.com/infection/infection/blob/master/src/TestFramework/PhpUnit/Adapter/PestAdapter.php#L105

For example first time it's called - return Pest version (to be displayed on console), second time it's called - return underlying PHPUnit version, to build configuration file.

It's dirty and hacky, however requires no changes in abstract files, which are really not needed for any test frameworks except Pest.

**Comment by theofidry** [comment]
@maks-rafalko why no returning a combination within `getVersion(): string`? `Pest X; PHPUnit Y`? This way we have the info we need both as a dev and a user

**Comment by maks-rafalko** [comment]
> @maks-rafalko why no returning a combination within `getVersion(): string`? `Pest X; PHPUnit Y`? This way we have the info we need both as a dev and a user, since the underlying PHPUnit version matters to the user too

Because we need to display Pest version on terminal when Infection is executed and we need to use PHPUnit version in our code, when deciding whether to use new PHPUnit features or not.

https://github.com/infection/infection/blob/804227770afbb876b4cbd24552535ddc0cdd331a/src/TestFramework/PhpUnit/Config/Builder/InitialConfigBuilder.php#L112-L131

So in different cases we need different versions (Pest, PHPUnit).

**Comment by theofidry** [comment]
>Because we need to display Pest version on terminal when Infection is executed

I don't feel it's a problem to show:

```
Pest version: 1.2.3 (PHPUnit 9.5.6)
```

For the config builder it's harder to test, but I imagine we could actually have a PestConfigBuilder than can parse the PHPUnit version from the string and then leverage the PHPUnit builder?
